Add docs for scaling private locations (#2582)
elastic/observability-docs: docs/en/observability/synthetics-private-location.asciidoc

A private location should be set up against an agent policy that runs on a single {agent}.
Do _not_ run the same agent policy on multiple agents being used for private locations, as you may
end up with duplicate or missing tests. Private locations do not currently load balance tests across
multiple {agents}. See <<synthetics-private-location-scaling>> for information on increasing the capacity
within a private location.

> Any private location can only run two browser tests simultaneously, and as many lightweight checks as resources allow.

Reviewer: Use the same language you use in the section below.

```suggestion
By default private locations are configured to allow two simultaneous browser tests and an unlimited number of lightweight checks.
```

Now you can add monitors to your private location in <<synthetics-get-started-ui,the {uptime-app}>> or using the <<synthetics-get-started-project,Elastic Synthetics library's `push` method>>.

TIP: You can run the same monitor both from Elastic's global managed infrastructure _and_ your private locations to compare results.

[discrete]
[[synthetics-private-location-scaling]]
= Scaling private locations

By default private locations are configured to allow two simultaneous browser tests, and an unlimited number of lightweight checks. 
These limits can be set via the environment variables `SYNTHETICS_LIMIT_{TYPE}`, where `{TYPE}` is one of `BROWSER`, `HTTP`, `TCP`, and `ICMP` 

> for the container running the {{agent}} docker image. 

Reviewer: ```suggestion
for the container running the {agent} docker image. 
```

It is critical to allocate enough memory and CPU capacity to handle configured limits. 

> Start by allocating at 2 GiB of memory and two cores per browser instance to ensure consistent

Reviewer 2: Typo?
>Start by allocating at 2 GiB of memory

Start by allocating at _least_ 2 GiB of memory

Reviewer: ```suggestion
Start by allocating at least 2 GiB of memory and two cores per browser instance to ensure consistent
```

> A good rule of thumb is to allocate at 2 GiB of memory and 2 cores per browser instance in order to have more consistent
> performance and avoid out of memory errors. Resource requirements will vary depending on workload. 

Reviewer: I'm not sure how well "rule of thumb" translates.

```suggestion
Start by allocating at 2 GiB of memory and two cores per browser instance to ensure consistent
performance and avoid out-of-memory errors. Then adjust as needed. Resource requirements will vary depending on workload. 
```

> Much less memory is needed for lightweight monitors. Start by allocating at 512MiB of memory and two cores for 

Reviewer 2: Same? (at _least_...)

Reviewer: ```suggestion
Much less memory is needed for lightweight monitors. Start by allocating at least 512MiB of memory and two cores for 
```

Author: I really wonder how I made this same typo twice 🤔

> For lightweight monitors much less memory is needed. We recommend starting at 512MiB of memory and two cores for 
> lightweight checks and increasing allocated memory and CPU based on observed usage patterns.

Reviewer: Use the same structure as above. 

```suggestion
Much less memory is needed for lightweight monitors. Start by allocating at 512MiB of memory and two cores for 
lightweight checks. Then increase allocated memory and CPU based on observed usage patterns.
```

These limits are for simultaneous tests, not total tests. For example, if
60 browser tests were scheduled to run once per hour and each took 1 minute to run, that would fully occupy one execution slot.
However, it is a good practice to set up execution slots with extra capacity. A good starting point would be to over-allocate by

> a factor of 5. In the previous example that would mean mean allocating 5 slots.

Reviewer 2: Typo _mean mean_

Reviewer: ```suggestion
a factor of 5. In the previous example that would mean allocating 5 slots.
```

> Note that these limits are for simultaneous tests, not total tests. If, for instance,
> 60 browser tests were scheduled to run once per hour, with each taking exactly 1 minute to run that would fully occupy one execution slot.
> That being said, it is not a good practice to leave execution slots without extra capacity. A good starting point would be to over-allocate by
> a factor of 5. In the previous example that would mean mean allocating 5 slots.

Reviewer: ```suggestion
These limits are for simultaneous tests, not total tests. For example, if
60 browser tests were scheduled to run once per hour and each took 1 minute to run, that would fully occupy one execution slot.
However, it is a good practice to set up execution slots with extra capacity. A good starting point would be to over-allocate by
a factor of 5. In the previous example that would mean mean allocating 5 slots.
```
